src/interfaces/common.ts

```diff
@@ -94,12 +94,7 @@ export interface Config {
 		 * globs that are used in the `mapEnvironment()` function to match a given environment with one
 		 * of the valid Jira environment types.
 		 */
-		environmentMapping?: {
-			development?: string[];
```

Comment:
This mapping is supplied by user during runtime, we don't know for sure

Reply:
We don't know it but that's exactly what the mapping does. Jira only accepts these 4, so if we can't match any of them, then Jira will reject the payload.

Reply:
On push webhook, the config of the mappings is saved to RepoSyncState [from here](https://github.com/atlassian/github-for-jira/blob/main/src/services/user-config-service.ts#L123), which takes a free form. 

On deployment webhook, this config is read. But [further map here](https://github.com/atlassian/github-for-jira/blob/main/src/transforms/transform-deployment.ts#L247).

So in short, the current behaviour is on saving the config, we take anything user provides and keep the ordering. When use it, we map it according to what user provide/order and set to "unmapped" if the result is not a valid jira env state.

---
Let me know if above make sense. And btw, the typings here doesn't reflect the "free form" part during push webhook, so I want to change it to match it.

---
Alternatively we can restrict from "free form" and only accept "development/prod/test/etc" on push webhook, but then we need to deal with existing data and backfill them, too much trouble.